install/kubernetes/helm/istio/charts/prometheus/templates/configmap.yaml

```diff
@@ -251,19 +251,62 @@ data:
         action: replace
         target_label: kubernetes_name
 
-    # Example scrape config for pods
-    - job_name: 'kubernetes-pods'
+    - job_name: 'kubernetes-pods-non-istio'
       kubernetes_sd_configs:
       - role: pod
+      relabel_configs:  # If first two labels are present, pod should be scraped  by the istio-secure job.
+      - source_labels: [__meta_kubernetes_pod_annotation_sidecar_istio_io_status]
+        action: drop
+        regex: (.+)
+      - source_labels: [__meta_kubernetes_pod_annotation_istio_mtls]
+        action: drop
+        regex: (true)
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
+        action: keep
+        regex: true
+      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
+        action: replace
+        target_label: __metrics_path__
+        regex: (.+)
+      - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
+        action: replace
+        regex: ([^:]+)(?::\d+)?;(\d+)
+        replacement: $1:$2
+        target_label: __address__
+      - action: labelmap
+        regex: __meta_kubernetes_pod_label_(.+)
+      - source_labels: [__meta_kubernetes_namespace]
+        action: replace
+        target_label: namespace
+      - source_labels: [__meta_kubernetes_pod_name]
+        action: replace
+        target_label: pod_name
 
+    - job_name: 'kubernetes-pods-istio-secure'
+      scheme: https
+      tls_config:
+        ca_file: /etc/istio-certs/root-cert.pem
+        cert_file: /etc/istio-certs/cert-chain.pem
+        key_file: /etc/istio-certs/key.pem
+        insecure_skip_verify: true  # prometheus does not support secure naming.
+      kubernetes_sd_configs:
+      - role: pod
       relabel_configs:
+      # sidecar status annotation is added by sidecar injector and
+      # istio_workload_mtls_ability can be specifically placed on a pod to indicate its ability to receive mtls traffic.
+      - source_labels: [__meta_kubernetes_pod_annotation_sidecar_istio_io_status, __meta_kubernetes_pod_annotation_istio_mtls]
```

Comment:
question: should we order this filter after the one that checks for `prometheus.io/scrape`?  Which do we expect to be more selective?  It feels like the `prometheus.io/scrape` may be more selective and cheaper (no regex).

Reply:
Good point, moved the `scrape` annotation above the other.

Reply:
How can be this done reliably? What if sidecar is added, you have the label, but we make the authn mtls policy to be configured as plaintext?

Reply:
The `istio_mtls` annotation was added for that reason, but never implemented.
plaintext *is* a mode of failure.